app/javascript/components/enrollment/steps/Exposure.js

```diff
@@ -165,24 +136,21 @@ class Exposure extends React.Component {
     const modified = this.state.modified;
     this.setState(
       {
-        current: { ...current, patient: { ...current.patient, first_positive_lab_at: first_positive_lab.specimen_collection }, first_positive_lab },
-        modified: { ...modified, patient: { ...modified.patient, first_positive_lab_at: first_positive_lab.specimen_collection }, first_positive_lab },
+        current: { ...current, patient: { ...current.patient, first_positive_lab_at: first_positive_lab?.specimen_collection }, first_positive_lab },
+        modified: { ...modified, patient: { ...modified.patient, first_positive_lab_at: first_positive_lab?.specimen_collection }, first_positive_lab },
         showLabModal: false,
       },
       () => {
         this.props.setEnrollmentState({ ...this.state.modified });
       }
     );
+    this.updateSOandPosLabValidations(current.patient, first_positive_lab);
```

Comment:
Is there a reason this isn't in the callback of the setState?

Reply:
it doesn't need to be in the callback of setState because `current.patient` is not modified by setState (only `current.first_positive_lab` is), and `first_positive_lab` is the argument of the parent function, so neither of the arguments rely on the state update to be finished

Reply:
https://github.com/SaraAlert/SaraAlert/pull/861/commits/76f2c63428e7bc885107fc4ba918068725877eea#diff-238f84238ca8a6fee3e8ebb55563658dd6716d2cf063b2bcf655b1b37de161fcR145